### ISSUE TITLE
os._exit is incorrectly flagged as a "protected-access" issue

### ISSUE THREAD

**Issue by ngie-eign** [PyCQA/pylint#6473]
`os._exit` is a supported function, matching the Unix libcall, `_exit`. It should not be flagged, but is incorrectly flagged by pylint as a W0212 (protected-access) issue.

**Comment by jacobtylerwalls** [comment]
I think there's value in leaving the warning. We don't want to get involved deciding which underscored methods are nevertheless allowable to call. The docs indicate this is [usually not to be called](https://docs.python.org/3/library/os.html?highlight=_exit#os._exit), other than cleaning up after a `fork()`. That means on balance we should warn about its usage, no?

**Comment by ngie-eign** [comment]
> I think there's value in leaving the warning. We don't want to get involved deciding which underscored functions are nevertheless allowable to call. The docs indicate this is [usually not to be called](https://docs.python.org/3/library/os.html?highlight=_exit#os._exit), other than cleaning up after a `fork()`. That means on balance we should warn about its usage, no?

Generally speaking, post-fork(), but there are other instances where os._exit should be used to avoid calling atexit handlers, flushing I/O streams, etc.

Either way, os._exit is a false positive in this case that should be handled differently than it currently is. I realize there needs to be a minimum bar of understanding when using APIs like fork/_exit, but (selfishly) I would like to decrease the barrier of entry so folks who are more junior or less aware of systems programming can trust tools like pylint to help inform them of items which should be handled in code.

**Comment by jacobtylerwalls** [comment]
> Either way, os._exit is a false positive in this case that should be handled differently than it currently is.

I think this is a classic use case for `exclude-protected` in your `pylintrc`. Regarding adding it to the default `pylintrc` generated by `--generate-rcfile`, do you think the false positives outweigh the false negatives from people incorrectly calling `_exit` such that we should add it to the list?

**Comment by Pierre-Sassoulas** [comment]
Adding this issue in the 2.15 milestone so we take a decision regarding ``os._exit`` being in the default value of ``exclude-protected`` by then.

**Comment by DanielNoord** [comment]
@jacobtylerwalls  @Pierre-Sassoulas Any opinion about this?

**Comment by Pierre-Sassoulas** [comment]
I say let's put this in default value, one user want it and took the time to ask for it. Also the documentation clearly explain the context in which you need to call it, I don't think there's an alternative for "impossible to catch sys exit" : https://docs.python.org/3/library/os.html#os._exit

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas I just saw that `exclude-protected` only stores attribute names. So if we add `_exit` any `MyClass()._exit` also won't flag anymore. Is that something we want to do?

**Comment by Pierre-Sassoulas** [comment]
Right, no. If we need to change the way exclude protected work it can probably wait.

**Comment by DanielNoord** [comment]
I mean users can add `_exit` themselves, but I'm not sure we should add it to the default values as that would make us ignore all `_exit` implementations. `exit` seems like an often used word.

Perhaps close as won't fix then?

**Comment by Pierre-Sassoulas** [comment]
I agree we should add ``os._exit`` in the default value but not ``_exit``. It means changing the way the check work (and also affect performance by using the qname and inference instead of simply the attribute name ?). So it's a lot of work. Should we create an issue for that and revisit later when this is easier to do ?

**Comment by DanielNoord** [comment]
Are we sure we want to go through the effort of creating a new option? Maintaining `exclude-protected` and `exclude-protected-qname` (temporary name obviously) seems like quite a bit of effort for one single edge case.

**Comment by Pierre-Sassoulas** [comment]
I think the use case for exclude protected will always be "this class an attribute combination is permitted", so we could probably replace the current ``exclude-protected`` by ``exclude-protected-qname`` entirely and call that an enhancement. It's not worth it right now but tracking the issue might be ?

**Comment by jacobtylerwalls** [comment]
That would be a valuable enhancement and allow us to provide these one-off defaults.

**Comment by Pierre-Sassoulas** [comment]
Blocked by https://github.com/PyCQA/pylint/issues/7343

**Comment by Pierre-Sassoulas** [comment]
Closing as it's fixed in #8364 , thank you @mbyrnepr2 :)
